src/common/non_file_printing_counter.hpp

```diff
@@ -0,0 +1,19 @@
+/**
+ * @file non_file_printing_counter.hpp
+ * @author Radek Vana
+ * @brief
+ * @date 2021-04-29
+ */
+
+#include <cstdint>
+
+class NonFilePrintingCounter {
+    static unsigned counter;
```

Comment:
This looks like a singleton, but that probably wasn't the intent. From the usage of this class it looks more like a guard combined with the guard variable into one piece. Looks overengineered to me.

Reply:
this is not singleton